src/Microsoft.AspNetCore.Http.Abstractions/HostString.cs

```diff
@@ -38,6 +64,92 @@ public bool HasValue
         }
 
         /// <summary>
+        /// Returns the value of the domain part of the host. The port is removed if it was present.
+        /// IPv6 addresses will have brackets added if they are missing.
+        /// </summary>
+        /// <returns></returns>
+        public string Domain
+        {
+            get
+            {
+                if (string.IsNullOrEmpty(_value))
+                {
+                    return string.Empty;
+                }
+
+                // normalize the value
+                var uriComponent = ToUriComponent();
+
+                int index;
+                if ((index = uriComponent.IndexOf(']')) >= 0)
+                {
+                    // optimization if uriComponent is exactly the domain
```

Comment:
FYI `Substring()` does this optimization for you already. Won't allocate in the whole string case.